### PR TITLE
End calendar focus

### PR DESCRIPTION
**Pull request by AlekhyaYalla**

### CODE REVIEW

File: ui/react-components/components/CalendarPicker/CalendarPicker.jsx [hunk 1]
```diff
@@ -28,14 +28,18 @@ const CalendarPicker = (props) => {
...
+            <button data-testid="calendar-icon" onClick={handleOnClick}>
+                <i className={classNames("fa", "fa-calendar")} aria-hidden="true"></i>
+            </button>
+            {showDatePicker && <AppointmentDatePicker
                 isDisabled={isDisabled}
                 minDate={minDate}
                 value={date}
-                onChange={onChange} />}
+                onChange={onChange}
+                hide={() => {
```
Comment (bsneha90) [inline ui/react-components/components/CalendarPicker/CalendarPicker.jsx]: hide sounds more of a boolean value, than a function

File: ui/react-components/components/DatePicker/DatePicker.jsx [hunk 2]
```diff
@@ -15,17 +15,18 @@ import DatePickerCustomHeader from './DatePickerCustomHeader.jsx';
...
 
     let containerStyles=[];
     isDisabled && containerStyles.push(disable);
 
-    const handleBlurOfDateInput = (date) =>{
-        const valueEntered= datetoMomentDate(date);
-        onChange(valueEntered)
+    const handleBlurOfDateInput = (date) => {
+        const valueEntered = datetoMomentDate(date);
+        hide && hide();
```
Comment (bsneha90) [inline ui/react-components/components/DatePicker/DatePicker.jsx]: what happens if i focus on date input and then click on calendar date?

**Comment by bsneha90** [comment]
Can we hide the calendar on press of esc button instead of on enter?

**Comment by bsneha90** [comment]
Instead of making change in all layers, can we have an event listener for the keypress and do the required operation?

File: ui/react-components/components/AddAppointment/AddAppointment.jsx [hunk 3]
```diff
@@ -71,7 +71,7 @@ const AddAppointment = props => {
         service: null,
         serviceType: null,
         location: null,
-        speciality: undefined,
+        speciality: null,
```
Comment (bsneha90) [inline ui/react-components/components/AddAppointment/AddAppointment.jsx]: why is this changed to null?

File: ui/react-components/components/DatePicker/DatePicker.jsx [hunk 4]
```diff
@@ -35,10 +35,17 @@ const AppointmentDatePicker = (props) => {
...
                       minDate={minDate && minDate.toDate()}
                       isDisabled={isDisabled}
-                      onBlur={handleBlurOfDateInput}/>
+                      onBlur={handleBlurOfDateInput}
+                      hideDatePicker = {hideDatePicker}
+           />
            <div disabled={isDisabled} data-testid="datePicker-Calendar"> <DatePicker
                 selected={isDisabled? '' : selectedDate}
-                onChange={(date) =>onChange(datetoMomentDate(date))}
+                onSelect={(date) => {
```
Comment (bsneha90) [inline ui/react-components/components/DatePicker/DatePicker.jsx]: can we bring it out as a method and not have it inline?

Comment (bsneha90) [reply ui/react-components/components/DatePicker/DatePicker.jsx]: why  date.getTime() === selectedDate.getTime() is required?

File: ui/react-components/components/CalendarPicker/CalendarPicker.jsx (hunk 1 above)
Comment (AlekhyaYalla) [reply ui/react-components/components/CalendarPicker/CalendarPicker.jsx]: Changed

File: ui/react-components/components/DatePicker/DatePicker.jsx (hunk 2 above)
Comment (AlekhyaYalla) [reply ui/react-components/components/DatePicker/DatePicker.jsx]: Fixed

File: ui/react-components/components/AddAppointment/AddAppointment.jsx (hunk 3 above)
Comment (AlekhyaYalla) [reply ui/react-components/components/AddAppointment/AddAppointment.jsx]: this went in separte commit

File: ui/react-components/components/DatePicker/DatePicker.jsx (hunk 4 above)
Comment (AlekhyaYalla) [reply ui/react-components/components/DatePicker/DatePicker.jsx]: This is for, we are closing the date picker onChange of selected date. If i select the same date, onChange will not be called, in that case, we are checking if the same day is selected, it will close the date picker. I will move it to a separate function, not inline

Comment (bsneha90) [reply ui/react-components/components/DatePicker/DatePicker.jsx]: date.getTime() === selectedDate.getTime() will get time not date right?

File: ui/react-components/components/DatePicker/DatePicker.jsx [hunk 5]
```diff
@@ -40,12 +40,8 @@ const AppointmentDatePicker = (props) => {
            />
            <div disabled={isDisabled} data-testid="datePicker-Calendar"> <DatePicker
                 selected={isDisabled? '' : selectedDate}
-                onSelect={(date) => {
-                    selectedDate && date.getTime() === selectedDate.getTime() && hideDatePicker && hideDatePicker()}}
-                onChange={(date) =>{
-                    onChange(datetoMomentDate(date));
-                    hideDatePicker && hideDatePicker();
-                }}
+                onSelect={() => hideDatePicker && hideDatePicker()}
```
Comment (bsneha90) [inline ui/react-components/components/DatePicker/DatePicker.jsx]: I see change in logic??

Comment (bsneha90) [reply ui/react-components/components/DatePicker/DatePicker.jsx]: Can we have test to cover this?

File: ui/react-components/components/DatePicker/DateInput.jsx [hunk 6]
```diff
@@ -28,36 +28,44 @@ const DateInput = (props) =>{
...
     };
 
     const handleKeyDown= (e) => {
         if (e.key === 'Enter') {
-           inputRef.current.blur();
+            inputRef.current.blur();
+            const inputValue = e.target.value;
+            const isValidDate = validateDate({dateValue: inputValue, minDate, maxDate});
+            if (isValidDate)
+                hideDatePicker && hideDatePicker();
```
Comment (bsneha90) [inline ui/react-components/components/DatePicker/DateInput.jsx]: I think hideDatePicker does not make sense for all scenario's. Can we instead call it handleEnterKeyDown or something similar?

Comment (AlekhyaYalla) [reply ui/react-components/components/DatePicker/DateInput.jsx]: CHanged them

File: ui/react-components/components/DatePicker/DatePicker.jsx (hunk 5 above)
Comment (AlekhyaYalla) [reply ui/react-components/components/DatePicker/DatePicker.jsx]: yes, added
